pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -650,11 +650,11 @@ func TestPriorityQueue_NominatedPodsForNode(t *testing.T) {
 		t.Errorf("Expected: %v after Pop, but got: %v", highPriorityPodInfo.Pod.Name, p.Pod.Name)
 	}
 	expectedList := []*framework.PodInfo{medPriorityPodInfo, unschedulablePodInfo}
-	if !reflect.DeepEqual(expectedList, q.NominatedPodsForNode("node1")) {
-		t.Error("Unexpected list of nominated Pods for node.")
+	if diff := cmp.Diff(expectedList, q.NominatedPodsForNode("node1")); diff != "" {
```

Comment:
In addition to comparing the content, we should add some logic below to ensure the returned list is safe to mutate: e.g., modify the returned list, and verify if the original list stays unchanged.

Reply:
I added a simple logic to modify the list.

Reply:
You will have to also verify the modfiied list doesn't change the original data. Say, after the modification:

```go
if len(podInfos) == q.NominatedPodsForNode() {
    t.Errorf(...)
}
```

Reply:
You miss the logic to compare the mutated list with the original data:

```go
if diff := cmp.Diff(podInfos, q.NominatedPodsForNode("node1")); diff == "" {
    t.Error(...)
}
```